eclass/debian.eclass

```diff
@@ -0,0 +1,85 @@
+# Copyright 2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+# @ECLASS: debian.eclass
+# @MAINTAINER:
+# Sam James <sam@gentoo.org>
+# @SUPPORTED_EAPIS: 8
+# @BLURB: Convenience eclass for easily using Debian patchsets as upstream
+# @DESCRIPTION:
+# TODO
+
+# @ECLASS_VARIABLE: DEBIAN_PN
+# @PRE_INHERIT
+# @DESCRIPTION:
+# Debian name for the package. Defaults to ${PN}.
+: ${DEBIAN_PN:=${PN}}
+
+# @ECLASS_VARIABLE: DEBIAN_PV
+# @PRE_INHERIT
+# @DESCRIPTION:
+# Debian patch level. Defaults to ${PV/*_p}.
+: ${DEBIAN_PV:=${PV/*_p}}
+
+# @ECLASS_VARIABLE: DEBIAN_ORIG
+# @DEFAULT_UNSET
+# @PRE_INHERIT
+# @DESCRIPTION:
+# Controls whether the "original" upstream sources (unpatched) are fetched
+# from Debian or not. Defaults to no.
+
+# @ECLASS_VARIABLE: DEBIAN_ORIG_PV
+# @PRE_INHERIT
+# @DESCRIPTION:
+# Controls the PV used without the patchlevel. Defaults to ${PV/_p*}.
+: ${DEBIAN_ORIG_PV:=${PV/_p*}}
+
+# @ECLASS_VARIABLE: DEBIAN_ORIG_EXT
+# @DEFAULT_UNSET
+# @PRE_INHERIT
+# @DESCRIPTION:
+# Controls the Debian "original" upstream sources (unpatched) file extension.
+# Defaults to gz.
+: ${DEBIAN_ORIG_EXT:=gz}
```

Comment:
Also, in the ebuild below you're using `.gz` (and I'd personally prefer including the dot in general).

Reply:
On the latter, this would be inconsistent with gnome.org's GNOME_TARBALL_SUFFIX, font-ebdftopcf.eclass's FONT_SUFFIX, font.eclass' FONT_SUFFIX, xorg-3.eclass's XORG_TARBALL_SUFFIX, etc.

Reply:
They all suck then? ;-)

What if someone published an uncompressed `.tar`? ;-)